graph/iterator/map.go

```diff
@@ -82,6 +95,28 @@ func (it *mapIter) node() graph.Node {
 	return *(*graph.Node)(mapiterelem(it.it))
 }
 
+// line returns the value of the iterator's current map entry.
+func (it *mapIter) line() graph.Line {
+	if it.it == nil {
+		panic("mapIter.node called before next")
```

Comment:
`mapIter.line` and similarly in `weightedLine`.

Reply:
Done. I'm looking forward to this being a single pair of generic functions.